packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -217,52 +241,159 @@ impl Abigen {
         })
     }
 
+    /// Expand a predicate into type-safe Rust bindings based on its ABI. See `expand_contract` for
+    /// more details.
+    pub fn expand_predicate(&self) -> Result<TokenStream, Error> {
+        let name = ident(&self.name);
+        let name_mod = ident(&format!("{}_mod", self.name.to_string().to_snake_case()));
+
+        let includes = self.includes();
+
+        let encode_data_function = self.main_function()?;
+        let code = if self.no_std {
+            quote! {}
+        } else {
+            quote! {
+                #[derive(Debug)]
+                pub struct #name{
+                    address: Bech32Address,
+                    code: Vec<u8>,
+                    data: UnresolvedBytes
+                }
+
+                impl #name {
+                    pub fn new(code: Vec<u8>) -> Self {
+                        let address: Address = (*Contract::root_from_code(&code)).into();
+                        Self {
+                            address: address.into(),
+                            code,
+                            data: UnresolvedBytes::new()
+                        }
+                    }
+
+                    pub fn load_from(file_path: &str) -> Result<Self, SDKError> {
+                        Ok(Self::new(std::fs::read(file_path)?))
+                    }
+
+                    pub fn address(&self) -> &Bech32Address {
+                        &self.address
+                    }
+
+                    pub fn code(&self) -> Vec<u8> {
+                        self.code.clone()
+                    }
+
+                    pub fn data(&self) -> UnresolvedBytes {
+                        self.data.clone()
+                    }
+
+                    pub async fn receive_from_wallet(&self, wallet: &WalletUnlocked, amount:u64, asset_id: AssetId, tx_parameters: Option<TxParameters>) -> Result<(String, Vec<Receipt>), SDKError> {
+                        let tx_parameters = tx_parameters.unwrap_or(TxParameters::default());
+                        wallet
+                            .transfer(
+                                self.address(),
+                                amount,
+                                asset_id,
+                                tx_parameters
+                            )
+                            .await
+                    }
+
+                    pub async fn spend_to_wallet(&self, wallet: &WalletUnlocked, amount:u64, asset_id: AssetId, tx_parameters: Option<TxParameters>) -> Result<Vec<Receipt>, SDKError> {
```

Comment:
The same suggestion about shortening the function name here. `spend_to_wallet(&self, wallet: &WalletUnlocked)` mentions "wallet" three times. 

```Rust
fn spend(&self, to: &WalletUnlocked) [...]
```

The above would achieve the same semantics but a lot more concisely!